backend/benefit/applications/api/v1/application_batch_views.py

```diff
@@ -60,6 +70,34 @@ class ApplicationBatchViewSet(AuditLoggingModelViewSet):
         "applications__company_contact_person_email",
     ]
 
+    def get_serializer_class(self):
+        """
+        ApplicationBatchSerializer for default behaviour on mutation functions,
+        ApplicationBatchListSerializer for listing applications on a batch
+        """
+        if self.action in ["create", "update", "partial_update", "destroy"]:
+            return ApplicationBatchSerializer
+
+        return ApplicationBatchListSerializer
+
+    def get_batch(self, id: str) -> ApplicationBatch:
+        """
+        Just a wrapper for Django's get_object_or_404 function
+        """
+        return get_object_or_404(ApplicationBatch, id=id)
+
+    @transaction.atomic
+    def destroy(self, request, pk=None):
+        """
+        Override default destroy(), batch can only be deleted if it's status is "draft"
+        """
+        batch = self.get_batch(pk)
+        if batch.status == ApplicationBatchStatus.DRAFT:
+            batch.delete()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        else:
+            return Response(status=status.HTTP_406_NOT_ACCEPTABLE)
```

Comment:
I think 400 would be a better fit also. MDN says about 406:  "In practice, this error is very rarely used. Instead of responding using this error code, which would be cryptic for the end user and difficult to fix, servers ignore the relevant header and serve an actual page to the user. It is assumed that even if the user won't be completely happy, they will prefer this to an error code."